plugins/SegmentEditor/API.php

```diff
@@ -107,16 +107,7 @@ protected function checkAutoArchive($autoArchive, $idSite)
         }
 
         if ($autoArchive) {
-            if ($realTimeSegmentsEnabled && !Piwik::isUserHasAdminAccess($idSite)) {
-                // pre-processed segment for a given website requires admin access
-                throw new Exception(
-                    "Please contact Support to make these changes on your behalf. ".
-                    " To modify a pre-processed segment, a user must have admin access or super user access. "
-                );
-            } elseif (!$realTimeSegmentsEnabled) {
-                // we require view access only when only pre processed can be created
-                Piwik::checkUserHasViewAccess($idSite);
-            }
+            Piwik::checkUserHasViewAccess($idSite);
```

Comment:
I think it should be possible to set `autoArchive=1` only when browser archiving is disabled? Otherwise the segment would be never archived?

We'll also need to adjust the config text in `enable_create_realtime_segments` as it no longer restricts which kinds of users can create segments to be archived on-demand. Could later if needed add a new setting like `require_admin_access_for_preprocessed_segments=0` but not needed for now.

Reply:
> I think it should be possible to set autoArchive=1 only when browser archiving is disabled?

I think you mean when cron archiving isn't set up, but I don't think there's a good way to check for that? If browser archiving is enabled and cli archiving is enabled, then it should be possible to use auto archive.

Reply:
I'm not sure we should see having both browser archiving and cli archiving enabled as a use case? Of course few users might do it (eg because they forget to disable browser archiving) but it should be rather edge cause and most of the times there wouldn't be much of a benefit of having both running AFAIK. Unless there would be some?